### ISSUE TITLE
When waitForText\see fails mocha-junit-reporter have rubbish error message

### ISSUE THREAD

**Issue by aldarund** [Codeception/CodeceptJS#180]
It gives zero info about where and what failed.
```
<failure>
<![CDATA[
Error at Object.<anonymous> (node_modules/codeceptjs/lib/helper/WebDriverIO.js:997:51) at Object.exec (node_modules/webdriverio/build/lib/helpers/safeExecute.js:28:24) at Object.resolve (node_modules/webdriverio/build/lib/webdriverio.js:188:29) at node_modules/webdriverio/build/lib/webdriverio.js:452:32 at _fulfilled (node_modules/webdriverio/node_modules/q/q.js:834:54) at self.promiseDispatch.done (node_modules/webdriverio/node_modules/q/q.js:863:30)
]]>
</failure>
```

**Comment by aldarund** [comment]
Another example of meaningless error message:
```
<failure>
<![CDATA[ Error: element not visible ]]>
</failure>
```

**Comment by aldarund** [comment]
Same low quality error message using mochawesome reporter

**Comment by DavertMik** [comment]
Thanks for the report. I will work on improving JUnit reporter

**Comment by DavertMik** [comment]
If you use `mocha-junit-reporter`, I need to say that it is written not in a really good manner. According to its source file it was not designed to show failure messages at all:

```js
  if (err) {
    var failureElement = {
      _cdata: this.removeInvalidCharacters(err.stack)
    };

    config.testcase.push({failure: failureElement});
  }
```

I.e. only stack trace is pushed to report. Anyway, I'm still working on improvements of error messages

**Comment by aldarund** [comment]
@DavertMik mochaawesome produce same error messages. Im using it only to integrate into circleci.

**Comment by DavertMik** [comment]
Thanks for pointing to mochawesome, I really like its look. 
I think the solution would be to add xml and html reporters into the core of codeceptjs to provide complete reports

**Comment by aldarund** [comment]
@DavertMik there another pretty nasty problem. test results from both junit and mochawesome not awlays generated. they like 3 times they are generated, one time not. Same code same test. Nothing changed. Was running on local selenium first on circleci with such behaviour now switched to saucelab and still same and i see following in saucelab ->
http://c2n.me/3AVOMQm

**Comment by aldarund** [comment]
@DavertMik i created issue for test results not generated. https://github.com/Codeception/CodeceptJS/issues/192

**Comment by DavertMik** [comment]
Ok, @aldarund how do you like this kind of JUnit output?

```xml

<?xml version="1.0" encoding="UTF-8"?>
<testsuites name="Mocha Tests" time="26.724" tests="5" failures="2">
  <testsuite name="Root Suite" timestamp="2016-08-06T23:43:50" tests="0" failures="0" time="0">
  </testsuite>
  <testsuite name="Testing Begins" timestamp="2016-08-06T23:43:50" tests="1" failures="1" time="6.916">
    <testcase name="Testing Begins: ANI testing" time="6.916" classname="ANI testing">
      <failure><![CDATA[expected web page to include "bogus text that is not there"

Scenario Steps:

- I.see("bogus text that is not there") at Test.<anonymous> (examples/absolutenet_test.js:7:5)
- I.grabTitle() at Test.<anonymous> (examples/absolutenet_test.js:6:23)
- I.amOnPage("http://www.absolutenet.com/") at Test.<anonymous> (examples/absolutenet_test.js:5:5)

]]></failure>
    </testcase>
  </testsuite>
  <testsuite name="GitHub" timestamp="2016-08-06T23:43:57" tests="3" failures="1" time="6.741">
    <testcase name="GitHub: search" time="3.022" classname="search">
    </testcase>
    <testcase name="GitHub: signin" time="2.212" classname="signin">
    </testcase>
    <testcase name="GitHub: register" time="1.507" classname="register">
      <failure><![CDATA[Field q not found by name|text|CSS|XPath

Scenario Steps:

- Within .js-signup-form: I.fillField("q", "aaa") at examples/github_test.js:29:7
- Within .js-signup-form: I.fillField("user[password]", "user@user.com") at examples/github_test.js:28:7
- Within .js-signup-form: I.fillField("user[email]", "user@user.com") at examples/github_test.js:27:7
- Within .js-signup-form: I.fillField("user[login]", "User") at examples/github_test.js:26:7

]]></failure>
    </testcase>
  </testsuite>
  <testsuite name="Yahoo test" timestamp="2016-08-06T23:44:20" tests="1" failures="0" time="13.067">
    <testcase name="Yahoo test: Nigthmare basic test" time="13.067" classname="Nigthmare basic test">
    </testcase>
  </testsuite>
</testsuites>
```

I think to include that in pull request

**Comment by aldarund** [comment]
@DavertMik looks good. way better than currently :) honestly im happy as long as it possible to determine at what step it fails

**Comment by DavertMik** [comment]
Should be fixed in #193
